File: shared/types/voice.ts
export interface VoiceSession {
  id: string;
  status: "idle" | "listening" | "thinking" | "speaking";
  started_at: string;
}

export interface TTSRequest {
  text: string;
  speaker_id?: string;
  language?: string;
}

export interface TTSResponse {
  audio_url: string;
  duration_ms: number;
}

export interface STTResult {
  text: string;
  language: string;
  confidence: number;
}

File: backend/app/models/voice.py
from pydantic import BaseModel
from typing import Optional


class VoiceSession(BaseModel):
    id: str
    status: str = "idle"
    started_at: str


class TTSRequest(BaseModel):
    text: str
    speaker_id: Optional[str] = None
    language: Optional[str] = "zh"


class TTSResponse(BaseModel):
    audio_url: str
    duration_ms: int


class STTResult(BaseModel):
    text: str
    language: str
    confidence: float
